### PR TITLE
fix: SOQL Builder saves blank files

### PR DESCRIPTION
**Pull request by floralan**
### What does this PR do?
This is an issue that only occurs on Code Builder where customers see an empty result file saved in their local machine. 

When users click `show local` to save a result file, we can see a permission error:
image (1)

In this PR, instead of using the default typescript `fs` to write a file, we use `vscode.workspace.fs`. This class can handle the writing/reading permissions on remote machines. 

### What issues does this PR fix or reference?
https://github.com/forcedotcom/try-code-builder-feedback/issues/200 
@W-12081772@

### Functionality Before
Users see a blank file from saving the result file on the SOQL builder.

### Functionality After
Users can see the real result file on their local machine. 

https://user-images.githubusercontent.com/42667832/228017471-88c2b856-4f78-41cc-851e-23be2e485bb1.mov

### CODE REVIEW

File: packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ describe('Query Data File Service', () => {
   afterEach(() => {
     // delete the query-results directory and its files.
     // @ts-ignore
-    fs.rmdirSync(testResultsDirPath, { recursive: true });
+    fs.rm(testResultsDirPath, { recursive: true });
```
Comment (floralan) [inline packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts:61]: `(node:488) [DEP0147] DeprecationWarning: In future versions of Node.js, fs.rmdir(path, { recursive: true }) will be removed. Use fs.rm(path, { recursive: true }) instead`

File: packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts [hunk 2]
```diff
@@ -58,7 +58,7 @@ describe('Query Data File Service', () => {
   afterEach(() => {
     // delete the query-results directory and its files.
     // @ts-ignore
-    fs.rmdirSync(testResultsDirPath, { recursive: true });
+    fs.rmSync(testResultsDirPath, { recursive: true });
```
Comment (floralan) [inline packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts:61]: `(node:488) [DEP0147] DeprecationWarning: In future versions of Node.js, fs.rmdir(path, { recursive: true }) will be removed. Use fs.rm(path, { recursive: true }) instead`

Comment (floralan) [reply packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts:61]: This is not a part of the fix, but something I noticed when looking at the test history.

**Review by jgellin-sf — APPROVED** [review]

**Review by jonnyhork — APPROVED** [review]
Looks good to me!

File: packages/salesforcedx-vscode-soql/test/jest/queryDataFileService.test.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
```
Comment (randi274) [inline packages/salesforcedx-vscode-soql/test/jest/queryDataFileService.test.ts:2]: Small nit - 2023!

File: scripts/setup-jest.ts [hunk 4]
```diff
@@ -67,6 +67,7 @@ const getMockVSCode = () => {
       setStatusBarMessage: jest.fn(),
       withProgress: jest.fn(),
       createOutputChannel: jest.fn(),
+      showSaveDialog: jest.fn(),
```
Comment (floralan) [inline scripts/setup-jest.ts:70]: Add `showSaveDialog` and `fs` in vs code stub.

File: packages/salesforcedx-vscode-soql/test/vscode-integration/testUtilities.ts [hunk 5]
```diff
@@ -28,7 +28,7 @@ const soqlExtension = vscode.extensions.getExtension(
   'salesforce.salesforcedx-vscode-soql'
 );
 const soqlExports = soqlExtension?.exports;
-const { workspaceContext, channelService } = soqlExports;
+const { workspaceContext = {}, channelService = {} } = soqlExports ?? {};
```
Comment (floralan) [inline packages/salesforcedx-vscode-soql/test/vscode-integration/testUtilities.ts:31]: A quick fix for an edge case, but not been used in this PR.

File: packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts [hunk 6]
```diff
@@ -78,7 +78,7 @@ export class QueryDataFileService {
       // use .fsPath, not .path to account for OS.
       selectedFileSavePath = fileInfo.fsPath;
       // Save query results to disk
-      fs.writeFileSync(selectedFileSavePath, fileContent);
+      await vscode.workspace.fs.writeFile(fileInfo, fileContent);
```
Comment (floralan) [inline packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts:81]: I added `await` here to run it synchronously. @randi274

File: packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts [hunk 7]
```diff
@@ -80,7 +80,7 @@ describe('Query Data File Service', () => {
     expect(jsonFileService.getDataProvider()).instanceOf(JsonDataProvider);
   });
 
-  it('should save json file to disk on save', async () => {
+  it.skip('should save json file to disk on save', async () => {
     const jsonFileService = new TestFileService(
```
Comment (floralan) [inline packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts:84]: @randi274 I want to skip these two tests as we have the unit tests covering the `save()` function. 

I did another end-to-end test on Mac and Code Builder to confirm this fix is working for both. 

Another reason for skipping these two tests is this [error](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/4559036019/jobs/8042546932#:~:text=1)%20Query%20Data,ts%3A114%3A27)), which I spent 1 hr to debug but didn't have luck. Considering the limited time I have, I will create a WI so we can cycle back on this after GA.

Comment (floralan) [reply packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts:84]: @randi274 Sounds good. Thank you for doing the QA!

**Review by randi274 — CHANGES_REQUESTED** [review]
Ok one more small change and we should be good to go!

File: packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts [hunk 8]
```diff
@@ -79,44 +79,4 @@ describe('Query Data File Service', () => {
     );
     expect(jsonFileService.getDataProvider()).instanceOf(JsonDataProvider);
   });
-
-  it('should save json file to disk on save', async () => {
```
Comment (randi274) [inline packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts:83]: Reference for deleting this and making follow-up tests: W-12779641
